brainiak/funcalign/srm.py

```diff
@@ -425,3 +426,241 @@ def _srm(self, data):
                 logger.info('Objective function %f' % loglike)
 
         return sigma_s, w, mu, rho2, shared_response
+
+
+class DetSRM(BaseEstimator, TransformerMixin):
+    """Deterministic Shared Response Model (DetSRM)
+
+    Given multi-subject data, factorize it as a shared response S among all
+    subjects and an orthogonal transform W per subject:
+
+    .. math:: X_i \\approx W_i S ,~for~all~i=1\dots N
+
+    Parameters
+    ----------
+
+    n_iter : int, default: 10
+        Number of iterations to run the algorithm.
+
+    features : int, default: 50
+        Number of features to compute.
+
+    rand_seed : int, default: 0
+        Seed for initializing the random number generator.
+
+    verbose : boolean, default: False
+        Verbose mode flag.
+
+
+    Attributes
+    ----------
+
+    w_ : list of array, element i has shape=[voxels_i, features]
+        The orthogonal transforms (mappings) for each subject.
+
+    s_ : array, shape=[features, samples]
+        The shared response.
+
+    .. note::
+       The number of voxels may be different between subjects. However, the
+       number of samples must be the same across subjects.
+
+       The Deterministic Shared Response Model is approximated using the
+       Block Coordinate Descent (BCD) algorithm proposed in [Chen2015]_.
```

Comment:
That creates a link in the HTML version:
http://pythonhosted.org/brainiak/brainiak.funcalign.html